In [11]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression

import pandas as pd

In [5]:
data = pd.DataFrame({
    "Client_ID": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
    "Employment_Type": ["Full-time", "Part-time", "Unemployed", "Full-time", "Part-time",
                        "Unemployed", "Full-time", "Part-time", "Unemployed", "Full-time"],
    "Marital_Status": ["Single", "Married", "Divorced", "Single", "Married",
                       "Single", "Divorced", "Married", "Single", "Divorced"],
    "Credit_Score": [750, 680, 600, 720, 710, 580, 770, 690, 620, 740]
})
data

,Client_ID,Employment_Type,Marital_Status,Credit_Score
0,1,Full-time,Single,750
1,2,Part-time,Married,680
2,3,Unemployed,Divorced,600
3,4,Full-time,Single,720
4,5,Part-time,Married,710
5,6,Unemployed,Single,580
6,7,Full-time,Divorced,770
7,8,Part-time,Married,690
8,9,Unemployed,Single,620
9,10,Full-time,Divorced,740


In [3]:
df = pd.get_dummies(data, columns=["Employment_Type", "Marital_Status"], drop_first=True, dtype='int')

print(df)

   Client_ID  Credit_Score  Employment_Type_Part-time  \
0          1           750                          0   
1          2           680                          1   
2          3           600                          0   
3          4           720                          0   
4          5           710                          1   
5          6           580                          0   
6          7           770                          0   
7          8           690                          1   
8          9           620                          0   
9         10           740                          0   

   Employment_Type_Unemployed  Marital_Status_Married  Marital_Status_Single  
0                           0                       0                      1  
1                           0                       1                      0  
2                           1                       0                      0  
3                           0                       0   

In [4]:
df.columns

Index(['Client_ID', 'Credit_Score', 'Employment_Type_Part-time',
       'Employment_Type_Unemployed', 'Marital_Status_Married',
       'Marital_Status_Single'],
      dtype='object')

In [6]:
X = df[['Employment_Type_Part-time', 'Employment_Type_Unemployed', "Marital_Status_Married", 'Marital_Status_Single']]
y = df['Credit_Score']

X.corr()

,Employment_Type_Part-time,Employment_Type_Unemployed,Marital_Status_Married,Marital_Status_Single
Employment_Type_Part-time,1.000000,-0.428571,1.000000,-0.534522
Employment_Type_Unemployed,-0.428571,1.000000,-0.428571,0.356348
Marital_Status_Married,1.000000,-0.428571,1.000000,-0.534522
Marital_Status_Single,-0.534522,0.356348,-0.534522,1.000000


In [ ]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)

In [15]:
model = LinearRegression()
model.fit(X_train, y_train)

y_pred = model.predict(X_test)

coefficients = pd.DataFrame({
    "Feature": X.columns,
    "Coefficient": model.coef_
}).sort_values(by="Coefficient", ascending=False)

print("\nFeature Importance:")
print(coefficients)


Feature Importance:
                      Feature  Coefficient
3       Marital_Status_Single        -20.0
0   Employment_Type_Part-time        -27.5
2      Marital_Status_Married        -27.5
1  Employment_Type_Unemployed       -155.0



Негативні коефіцієнти для змінних Marital_Status_Single та Marital_Status_Married свідчать про те, що ці категорії мають менший позитивний вплив на цільову змінну порівняно з базовою категорією Divorced. Особи, які перебувають у стані незаміжнього/неодруженого та одруженого стану, мають нижчі кредитні рейтинги (на 20 та 30 балів) порівняно з розлученими особами.

Негативні коефіцієнти для Employment_Type_Part-time та Employment_Type_Unemployed вказують на те, що ці категорії мають менший позитивний вплив на цільову змінну порівняно з базовою категорією Full-time. Наявність роботи на повний робочий день позитивно впливає на кредитний рейтинг, тоді як робота на неповний робочий день та безробіття знижують кредитний рейтинг на 28 та 155
балів.


